api/views.py

```diff
@@ -204,6 +204,24 @@ class MediaItemView(MediaItemMixin, generics.RetrieveUpdateAPIView):
     serializer_class = serializers.MediaItemDetailSerializer
 
 
+class MediaItemUploadView(MediaItemMixin, generics.RetrieveUpdateAPIView):
```

Comment:
No, because the primary key used to retrieve the endpoint is the MediaItem. Since upload endpoints are mapped via OneToOne fields, this is the only key passed to the API. I'll experiment with doing it the other way and report back.

Reply:
Results: it is too horribly ugly to try and re-work all the permissions classes to return the correct results when starting with an upload endpoint and the "inside out" nature of the query set (all upload endpoints whose media items are viewable by the current user) is very hard to express. I think it's easier like this since the "upload endpoint" is essentially a different view of a media item.

Reply:
fair nuff :-)